packages/core/builders/CalendarEvent/builder.ts

```diff
@@ -0,0 +1,284 @@
+    this.users = luckyUsers;
+  }
+
+  private getLuckyUsers(
+    users: User[],
+    bookingCounts: {
+      username: string | null;
+      bookingCount: number;
+    }[]
+  ) {
+    if (!bookingCounts.length) users.slice(0, 1);
+
+    const [firstMostAvailableUser] = bookingCounts.sort((a, b) => (a.bookingCount > b.bookingCount ? 1 : -1));
+    const luckyUser = users.find((user) => user.username === firstMostAvailableUser?.username);
+    return luckyUser ? [luckyUser] : users;
+  }
+
+  public async buildTeamMembers() {
+    this.teamMembers = await this.getTeamMembers();
+  }
+
+  private async getTeamMembers() {
+    // Users[0] its organizer so we are omitting with slice(1)
+    const teamMemberPromises = this.users.slice(1).map(async function (user) {
+      return {
+        id: user.id,
+        username: user.username,
+        email: user.email || "", // @NOTE: Should we change this "" to teamMemberId?
+        name: user.name || "",
+        timeZone: user.timeZone,
+        language: {
+          translate: await getTranslation(user.locale ?? "en", "common"),
+          locale: user.locale ?? "en",
+        },
+        locale: user.locale,
+      } as PersonAttendeeCommonFields;
+    });
+    return await Promise.all(teamMemberPromises);
+  }
+
+  public buildUIDCalendarEvent() {
+    if (this.users && this.users.length > 0) {
+      throw new Error("call buildUsers before calling this function");
+    }
+    const [mainOrganizer] = this.users;
+    const seed = `${mainOrganizer.username}:${dayjs(this.calendarEvent.startTime)
+      .utc()
+      .format()}:${new Date().getTime()}`;
+    const uid = translator.fromUUID(uuidv5(seed, uuidv5.URL));
+    this.calendarEvent.uid = uid;
+  }
+
+  public setLocation(location: CalendarEventClass["location"]) {
+    this.calendarEvent.location = location;
+  }
+
+  public setUId(uid: CalendarEventClass["uid"]) {
+    this.calendarEvent.uid = uid;
+  }
+
+  public setDestinationCalendar(destinationCalendar: CalendarEventClass["destinationCalendar"]) {
+    this.calendarEvent.destinationCalendar = destinationCalendar;
+  }
+
+  public setHideCalendarNotes(hideCalendarNotes: CalendarEventClass["hideCalendarNotes"]) {
+    this.calendarEvent.hideCalendarNotes = hideCalendarNotes;
+  }
+
+  public setDescription(description: CalendarEventClass["description"]) {
+    this.calendarEvent.description = description;
+  }
+
+  public setCancellationReason(cancellationReason: CalendarEventClass["cancellationReason"]) {
+    this.calendarEvent.cancellationReason = cancellationReason;
+  }
+
+  public buildRescheduleLink(originalBookingUId: string) {
+    if (!this.eventType) {
+      throw new Error("Run buildEventObjectFromInnerClass before this function");
+    }
+    const isTeam = !!this.eventType.teamId;
+
+    const queryParams = new URLSearchParams();
+    queryParams.set("rescheduleUid", `${originalBookingUId}`);
+    const rescheduleLink = `${process.env.BASE_URL}/${
```

Comment:
Base url is deprecated:
```suggestion
    const rescheduleLink = `${process.env.NEXT_PUBLIC_WEBAPP_URL}/${
```